squidDrone/src/types/error_types.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef SRC_TYPES_ERROR_TYPES_HPP_
+#define SRC_TYPES_ERROR_TYPES_HPP_
+
+#include <cstdint>
+
+namespace types{
+
+  ///
+  /// @brief A error type enum
+  ///
+  enum class HalError: std::uint16_t{
+    /// Hal function is working
+    WORKING,
+    /// Configuration returned an error
```

Comment:
In Architecture the type is defined as "int".

Reply:
No it isn't.
The picture below shows, that HalError is the return type of the method. Int is the type of the second parameter of the method
![Screenshot from 2020-03-30 19-50-45](https://user-images.githubusercontent.com/42242163/77944860-f2707e00-72bf-11ea-9857-2243ca231bee.png)

HalError is further defined in [this diagram](https://github.com/elheck/Squiddrone/wiki/UML-class-types)

Reply:
Never mind. You were right.
Fixed in [this](https://github.com/elheck/Squiddrone/pull/168/commits/ac2f52442da08a005a1ffa1979ff307899050930) commit